cli/cmd/install.go

```diff
@@ -206,35 +249,91 @@ func isRunAsRoot(values map[string]interface{}) bool {
 	return false
 }
 
-func render(w io.Writer, values *l5dcharts.Values, crds bool, valuesOverrides map[string]interface{}) error {
-
-	crdFiles := []*loader.BufferedFile{
+func renderCRDs(w io.Writer, values *l5dcharts.Values, valuesOverrides map[string]interface{}) error {
+	files := []*loader.BufferedFile{
 		{Name: chartutil.ChartfileName},
 	}
-	cpFiles := []*loader.BufferedFile{
-		{Name: chartutil.ChartfileName},
+
+	for _, template := range templatesCrdFiles {
+		files = append(files,
+			&loader.BufferedFile{Name: template},
+		)
+	}
+	if err := charts.FilesReader(static.Templates, l5dcharts.HelmChartDirCrds+"/", files); err != nil {
+		return err
 	}
 
-	if crds {
-		for _, template := range templatesCrdFiles {
-			crdFiles = append(crdFiles,
-				&loader.BufferedFile{Name: template},
-			)
-		}
-		if err := charts.FilesReader(static.Templates, l5dcharts.HelmChartDirCrds+"/", crdFiles); err != nil {
-			return err
-		}
-	} else {
-		for _, template := range templatesControlPlane {
-			cpFiles = append(cpFiles,
-				&loader.BufferedFile{Name: template},
-			)
-		}
-		if err := charts.FilesReader(static.Templates, l5dcharts.HelmChartDirCP+"/", cpFiles); err != nil {
+	var partialFiles []*loader.BufferedFile
+	for _, template := range charts.L5dPartials {
+		partialFiles = append(partialFiles,
+			&loader.BufferedFile{Name: template},
+		)
+	}
+
+	// Load all partial chart files into buffer
+	if err := charts.FilesReader(static.Templates, "", partialFiles); err != nil {
+		return err
+	}
+
+	// Create a Chart obj from the files
+	files = append(files, partialFiles...)
+	chart, err := loader.LoadFiles(files)
+	if err != nil {
+		return err
+	}
+
+	// Store final Values generated from values.yaml and CLI flags
+	chart.Values, err = values.ToMap()
+	if err != nil {
+		return err
+	}
+
+	vals, err := chartutil.CoalesceValues(chart, valuesOverrides)
+	if err != nil {
+		return err
+	}
+
+	fullValues := map[string]interface{}{
+		"Values": vals,
+		"Release": map[string]interface{}{
+			"Namespace": controlPlaneNamespace,
+			"Service":   "CLI",
+		},
+	}
+
+	// Attach the final values into the `Values` field for rendering to work
+	renderedTemplates, err := engine.Render(chart, fullValues)
+	if err != nil {
+		return fmt.Errorf("failed to render the template: %w", err)
+	}
+
+	// Merge templates and inject
+	var buf bytes.Buffer
+	for _, tmpl := range chart.Templates {
+		t := path.Join(chart.Metadata.Name, tmpl.Name)
+		if _, err := buf.WriteString(renderedTemplates[t]); err != nil {
 			return err
 		}
 	}
 
+	_, err = w.Write(buf.Bytes())
+	return err
+}
+
+func renderControlPlane(w io.Writer, values *l5dcharts.Values, valuesOverrides map[string]interface{}) error {
```

Comment:
renderControlPlane and renderCRDs are mostly identical.  I think they can be consolidated if you pass the `files` slice in as a function argument

Reply:
i've extracted the common parts into another function, `renderChartToBuffer`